p2p/transport/webrtc/internal/async/condvar.go

```diff
@@ -0,0 +1,46 @@
+package async
+
+import "sync"
+
+// CondVar is an edge-Signaled condition shared by multiple goroutines.  It is
+// analogous in effect to the standard condition variable (sync.Cond) but uses
+// a channel for signaling.
+//
+// To wait on the CondVar, call Wait to obtain a channel. This channel will be
+// closed the next time the CondVar method is executed. A fresh channel is
+// created after each call to CondVar, and is shared among all calls to Wait
+// that occur in the window between signals.
+//
+// A zero CondVar is ready for use but must not be copied after first use.
+type CondVar struct {
```

Comment:
Hmmm, the advantage is no more or smaller then any kind of utility struct. It's just a matter of separation of concerns?  I mean we could also add a channel and mutex directly to the struct where it is used, but to me that's more ugly then having it nice as a utility struct?

Reply:
As such going to close this one for now. Unless you have stronger opinions about this one.

That being said, if we manage to avoid the current use of that CondVar all together I am even more happier, as I don't think I'm super happy about it, but neither did I see an immediate way to remove such a thing from the implementation I worked upon.